locator/tablets.cc

```diff
@@ -449,6 +449,17 @@ size_t tablet_metadata::external_memory_usage() const {
     return result;
 }
 
+bool tablet_metadata::has_replica_on(host_id host) const {
+    for (auto&& [id, map] : _tablets) {
+        for (auto&& tablet : map.tablet_ids()) {
+            if (map.get_shard(tablet, host)) {
+                return true;
+            }
+        }
+    }
```

Comment:
With 100k tablets, budget 0.1ms -> 1ns/iteration. It's borderline, but likely achievable as it's straightforward linear memory.

Please schedule improvement of the data structure to maintain an index.

Reply:
There is an idea to keep node* in tablet_metadata, which will allow us to avoid lookups in the hot path.

This particular method is not used on the hot path.